lib/page_extractor.rb

```diff
@@ -2,6 +2,18 @@
 
 class PageExtractor
 
+  def self.extract_page(url)
```

Comment:
I think it's time to turn these methods into instance methods instead of class methods.